### PR TITLE
Removed Node modules & test files from production build.

### LINKED ISSUE #812: Adding Bundler for production build
- This would help in compiling multiple JS file into a single file.
- It creates a dependency relationship graph
- It improves the load time.
- The generated file would be present in `/dist`

### PR DESCRIPTION
**Pull request by vikhyat187**
Closes #812 

The task was to remove the dev dependencies from the production build & test cases.
1. I've used webpack to bundle the files and the output file would be stored in ```/dist``` folder.
2. The script to build is ```yarn run build```.
3. The script to run the bundle is ```yarn run startbuild```.

To analyze the bundle we can use https://chrisbateman.github.io/webpack-visualizer/

[stats1.txt](https://github.com/Real-Dev-Squad/website-backend/files/10239774/stats1.txt) **Before bundling the stats file size was 22MB.**

[stats.txt](https://github.com/Real-Dev-Squad/website-backend/files/10202818/stats.txt) **After removing node modules the size became 650KB**

Download the ```stats.txt``` file and rename to ```stats.json```, upload to above link to analyze bundle.

[Links referred](https://blog.bitsrc.io/6-tools-and-techniques-to-analyze-webpack-bundle-size-817337f8cf91)

before_node_modules

after_bundling

### CODE REVIEW

**Comment by vikhyat187** [comment]
> At what point are dev dependencies removed?

On building the bundled file.

**Review by MehulKChaudhari — APPROVED** [review]
LGTM.

**Comment by prakashchoudhary07** [comment]
@vikhyat187 This look good, but would want to test the changes before pushing it to DEV

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  resolve: {
+    modules: [path.resolve(__dirname, "node_modules")],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        exclude: [/node_modules/, /test/],
+      },
```
Comment (prakashchoudhary07) [inline webpack.config.js:19]: Do we remove tests as well,
Do we not run the tests after the build is done?

How is this generally do, like do tests run and then build takes place or do build run then test runs?

Comment (vikhyat187) [reply webpack.config.js:19]: Since the test supporting dependencies were dev and we were removing them while bundling, so test wouldn;t work.

**Comment by vikhyat187** [comment]
> > @vikhyat187 This look good, but would want to test the changes before pushing it to DEV
> 
> Can we deploy it in a separate branch and test it out

Yes sure.

**Comment by prakashchoudhary07** [comment]
How are we going to remove node modules when deploying in production?

**Comment by vikhyat187** [comment]
> How are we going to remove node modules when deploying in production?

"webpack-node-externals": this helps in removing the dev dependencies.

**Comment by prakashchoudhary07** [comment]
I think we should remove dev dependencies from the production, we will have to look into this, 
Either run some command to remove dev dependencies or only install dependencies in prod and then we can merge this PR

**Comment by vikhyat187** [comment]
> I think we should remove dev dependencies from the production, we will have to look into this, Either run some command to remove dev dependencies or only install dependencies in prod and then we can merge this PR

Will check this out and get back.

**Comment by vikhyat187** [comment]
> > I think we should remove dev dependencies from the production, we will have to look into this, Either run some command to remove dev dependencies or only install dependencies in prod and then we can merge this PR
> 
> Will check this out and get back.

https://docs.npmjs.com/cli/v8/commands/npm-prune this one doesn't install dev dependencies, lets try it out @prakashchoudhary07

**Comment by vikhyat187** [comment]
Closing this PR as, this doesn't improve any performance in backend server.

**Pull request closed without merging**